### PR TITLE
Ditch tmp table for formula

### PR DESCRIPTION
**Pull request by buildlackey**
- eliminate rate increase table in favor of computing auto tuned rate on the fly using the Gennadiy Formula.
    - add additional tests to verify the ability to recover rate ramp-up behavior once error rate falls below threshold after initially exceeding it.

### CODE REVIEW

File: ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java [hunk 1]
```diff
@@ -69,64 +60,26 @@ public ConstantStepWiseRateIncreaser(int rampPeriodMillisecs,
...
-        assert (tmpClockTimeToRateTable.get(numSteps).time == rampPeriodMillisecs);
-        assert (tmpClockTimeToRateTable.get(numSteps).rate == finalRate);
-
-
-        ImmutableList.Builder<Cell> cellBuilder = new ImmutableList.Builder<>();
-        clockTimeToRateTable = cellBuilder.addAll(tmpClockTimeToRateTable).build();
+        this.initRate = initRate;
+        this.finalRate = finalRate;
+        this.incrementIntervalMillisecs = incrementIntervalMillisecs;
+        this.rateIncrementPerStep = rateIncrementPerStep;
```
Comment (gndcshv) [inline ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java]: Ditch `rateIncrementPerStep` local variable and just have `this.rateIncrementPerStep = spread / numSteps;`

File: ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java [hunk 2]
```diff
@@ -58,7 +59,6 @@
     private final String esDocType;
     private final boolean isBulkWrite;
 
-    private static final ImmutableMap<String, String> EMPTY_PARAMS = ImmutableMap.of();
```
Comment (gndcshv) [inline ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java]: I'm guessing `import com.google.common.collect.ImmutableMap;` can go too.

File: ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java (hunk 1 above)
Comment (buildlackey) [reply ndbench-core/src/main/java/com/netflix/ndbench/core/util/ConstantStepWiseRateIncreaser.java]: woops. ok

File: ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java (hunk 2 above)
Comment (buildlackey) [reply ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsWriter.java]: yup

**Pull request merged**